src/main/java/com/vacation_bot/core/vacation/VacationService.java

```diff
@@ -0,0 +1,113 @@
+package com.vacation_bot.core.vacation;
+
+import com.vacation_bot.core.BaseService;
+import com.vacation_bot.domain.models.UserModel;
+import com.vacation_bot.domain.models.VacationModel;
+import com.vacation_bot.domain.models.VacationTotalModel;
+import com.vacation_bot.repositories.RepositoryFactory;
+import com.vacation_bot.spring.exception.RepositoryException;
+import org.springframework.stereotype.Service;
+
+import java.time.*;
+import java.time.temporal.ChronoUnit;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * Reserve a vacation to the user.
+ */
+@Service
+public class VacationService extends BaseService{
+
+    private static final int DEFAULT_VACATION_TOTAL_DAYS = 20;
+
+    private static final String USER_NOT_FOUND_MESSAGE = "USER NOT FOUND";
+
+    private int currentYear = Calendar.getInstance().get(Calendar.YEAR);
+
+    public VacationService(RepositoryFactory factory) {
+        super(factory);
+    }
+
+    public String createVacation(final String userName, final String _startDate, final String _endDate) {
+
+        LocalDate startDate = LocalDate.parse(_startDate);
+        LocalDate endDate = LocalDate.parse(_endDate);
+
+        //the days reserved by user
+        int period = (int) ChronoUnit.DAYS.between(startDate, endDate);
+
+        //found user by name or aliases
+        UserModel user = Optional.ofNullable(getUserModelRepository()
```

Comment:
Spring data mongodb query methods support Optional as return type. It'll have more sence to change the return type in the `UserModelRepository` for `findByNameOrAliases`: https://github.com/spring-projects/spring-data-examples/tree/master/jpa/java8